src/main/java/fact/features/muon/HoughTransform.java

```diff
@@ -91,7 +91,7 @@
     private double[] circle_x;
     private double[] circle_r;
 
-    public ArrayList<int[]>[] chid2circles = new ArrayList[Constants.NUMBEROFPIXEL];
+    public ArrayList<ArrayList<int[]>> chid2circles = new ArrayList<>();
```

Comment:
You could initialize the arraylists to the proper length(if you know it). Makes it somewhat more efficient.